packages/ec-core/src/operator/selector/lexicase.rs

```diff
@@ -95,3 +100,200 @@ where
+        #[strategy(make_pop(#pop_size))] mut population: Vec<TestIndividual>,
+        #[strategy(0..1usize)] score_to_increase: usize,
+    ) {
+        prop_assert!(pop_size > 0);
+        prop_assert_eq!(pop_size, population.len());
+
+        let (winning_label, winning_individual) =
+            make_winning_individual(&population, score_to_increase);
+
+        population.push(winning_individual);
+
+        let num_test_cases = population[0].test_results.results.len();
+        let lexicase = Lexicase::new(num_test_cases);
+        let mut rng = rand::thread_rng();
+
+        let selected = lexicase.select(&population, &mut rng).unwrap().genome();
+        prop_assert_eq!(selected, &winning_label);
+    }
+
+    // A function that creates a `proptest` `Strategy`` that creates a random
+    // population of individuals, each with two scores. The "genome" of each
+    // individual is just a string version of their pair of scores, i.e.,
+    // "[2, 3]".
+    fn make_pop(pop_size: usize) -> impl Strategy<Value = Vec<TestIndividual>> {
+        collection::vec([any::<u8>(), any::<u8>()], pop_size).prop_map(|pop_scores| {
+            pop_scores
+                .into_iter()
+                .map(|scores| EcIndividual::new(format!("{scores:?}"), TestResults::from(scores)))
+                .collect::<Vec<_>>()
+        })
+    }
+
+    // Constructs what is guaranteed to be the "winning" individual
+    // that lexicase selection will return regardless of the ordering
+    // of the test cases. This is generated by finding the largest
+    // values for each of the test cases, and incrementing one of
+    // them (as determined by `score_to_increase`) by one. This
+    // ensures that this individual will tie with the best on
+    // one score, and win on the other.
+    fn make_winning_individual(
+        population: &[TestIndividual],
+        score_to_increase: usize,
+    ) -> (String, TestIndividual) {
+        let first_max = largest_test_case_value(population, 0);
+        let second_max = largest_test_case_value(population, 1);
+        // After incrementing one of the values by 1, this is guaranteed to
+        // be "better" than any other score, and will thus be the value
+        // selected.
+        let mut winning_scores = [first_max, second_max];
+        winning_scores[score_to_increase] =
+            winning_scores[score_to_increase].checked_add(1).unwrap();
+        let winning_label = format!("{winning_scores:?}");
+        let winning_individual: TestIndividual =
+            EcIndividual::new(winning_label.clone(), TestResults::from(winning_scores));
+        (winning_label, winning_individual)
+    }
+
+    // Get the largest test case value in the given population from the test case
+    // specified by `index`.
+    fn largest_test_case_value(population: &[TestIndividual], index: usize) -> u16 {
+        population
+            .iter()
+            .map(|i| i.test_results.results[index])
```

Comment:
I see what you did here - using `[...]` indexing :joy:  ( but since it's in a test it's fine )